text/0000-named-blocks.md

```diff
@@ -0,0 +1,635 @@
+    title=ima title
+    I'm a block provided by the component layout template.
+  </div>
+</div>
+```
+
+It would also be possible to pass a component factory to the header
+block from `x-modal`'s layout:
+
+```
+// ember-component x-modal layout
+{{header title="ima title"
+         main=(component 'x-modal-inner-content')}}
+```
+
+##### Block form of Unified Renderable syntax: OPEN QUESTIONS:
+
+Is the `{{#header}}...{{/header}}` syntax OK?
+
+Also, what does this syntax look like for Glimmer `@arg`s?
+
+There are a few candidates thus far:
+
+```
+{{#@group as |a b c|}}
+  ...
+{{/@group}}
+
+{{# @group as |a b c|}}
+  ...
+{{/ @group}}
+
+{{#render @group as |a b c|}}
+  ...
+{{/render}}
+```
+
+`render` looks the nicest, but it would mean resurrecting the `#render`
+helper from the deprecated dead in order to fit this one syntactical
+corner case. It might be alright to just put up with this one ugly
+syntax and punt a nicer alternative (perhaps involving `render`) to a future RFC.
+
+### Classic single-block syntax: `main` and `else` args
+
+It would be unfortunate if component authors had to use different
+syntaxes for rendering named blocks vs the traditional "default"
+and "inverse" blocks provided by the classic single-block syntax.
+
+Hence, the blocks provided in classic single-block syntax should also
+be exposed as properties (Ember) and args (Glimmer), and should have
+conventional, meaningful names: instead of "default" (which is a
+bit misleading) and "inverse", we standardize on `main` and `else`.
+
+#### Glimmer Components: `@main` and `@else`
+
+Given Glimmer component invocation:
+
+```
+{{#fancy-if cond=trueOrFalse}}
+  True
+{{else}}
+  False
+{{/fancy-component}}
+```
+
+The component layout could be:
+
+```
+{{#if cond}}
+  {{@main}}
+{{else}}
+  {{@else}}
+{{/if}}
+```
+
+Note that angle-bracket syntax doesn't support passing in an
+inverse/else block, but the block provided to angle-bracket invocation
+would be passed in as `@main`.
+
+#### Ember Components: `main` and `else`
+
+For Ember, we can't suddenly start setting `main` and `else` properties
+on the component instances as this would be a breaking change, and
+`main` in particular is not an uncommon property name.
+
+We also shouldn't punt on this feature for Ember components for the
+following reasons/use cases:
+
+- [ember-elsewhere](https://github.com/emberjs/rfcs/pull/226#issuecomment-299949000)
+  (and other similar patterns) require having access to the opaque block
+  so that it can be stashed on a service and rendered elsewhere
+- wrapper components that forward args/properties/blocks to another
+  intenal component; blocks need to be accessible as properties in order
+  to pass them into another component (otherwise you'd have to use a
+  combinatoric mess of block syntax + `if hasBlock` checks to forward
+  blocks through to the inner component)
+
+I propose an opt-in API; any Ember Component that wants `main`/`else`
```

Comment:
Yes. Basically, you could create a component like this, and just extend any of your components from that one.